ton_client/src/abi/encode_message.rs

```diff
@@ -161,9 +162,20 @@ impl CallSet {
             )?
         };
 
+        let func = match u32::from_str_radix(&self.function_name, 16) {
```

Comment:
Better use general numbers encodeing rules: decimal or hex prefixed with `0x`. You can use `decode_abi_number` function to decode id

Reply:
Ok